### PR TITLE
Implement ufuncs for omnisci array

### PR DESCRIPTION
**Pull request by guilhermeleobas**
This pull request implements 73/87 ufuncs listed [here](https://numpy.org/doc/stable/reference/ufuncs.html).

Partially closes https://github.com/xnd-project/rbc/issues/99

### CODE REVIEW

**Comment by guilhermeleobas** [comment]
Obs: I did not include tests of the type `fn(<scalar>, <array>)` because of https://github.com/xnd-project/rbc/issues/81

File: rbc/typesystem.py [hunk 1]
```diff
@@ -1120,6 +1120,8 @@ def get_data_type(self):
 
     class Boolean8(nb.types.Boolean):
 
+        bitwidth = 8
```
[the remark this answers is not in the document]
Comment (pearu) [reply rbc/typesystem.py]: All tests pass without it. So removing this addition.

File: rbc/omnisci_backend/numpy_math.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+from .omnisci_array import Array, ArrayPointer
+from rbc.utils import get_version
+if get_version('numba') >= (0, 49):
+    from numba.core import extending, types
+else:
+    from numba import extending, types
+
+
+def determine_dtype(a, dtype):
+    if dtype is not None:
```
[the remark this answers is not in the document]
Comment (pearu) [reply rbc/omnisci_backend/numpy_math.py]: Applied.

File: rbc/typesystem.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (guilhermeleobas) [reply rbc/typesystem.py]:
```python
a = Array(size, 'bool')
```

We use `bitwidth` information when creating the array. See https://github.com/xnd-project/rbc/blob/95c4cea1817312ae142d3207a08be2c53258e655/rbc/omnisci_backend/omnisci_array.py#L57

**Pull request merged**